### DISCUSSION TITLE
Split `pikasay` into a separate package

### DISCUSSION THREAD

**Discussion by AKorezin** [actionless/pikaur#828] [Q&A] [answered]
## Summary

Is it possible to create a split package for `pikasay`?
It breaks my habit of auto-completing `pik` to `pikaur`.

**Comment by actionless** [comment]
this wouldn't really lead to nowhere, as there are other packages which have the same "prefix":

```console
$ pikaur -Ssq pik | grep '^pik' | grep -v pikaur
pika-backup
pikopixel
pike
pikalogy
pik-git
pika-backup-git
pikapika-bin
pikasso-git
pikatorrent-bin
pikchr-fossil
piknik
piknik-bin
piknik-git
pikture-git
```

in fish i'm just getting along with it with this:

```console
$ grep -R pikaur ~/.config/fish/
/home/lie/.config/fish/modules/arch.fish:abbr p 'pikaur'
/home/lie/.config/fish/modules/arch.fish:abbr syu  'pikaur -Syu'
/home/lie/.config/fish/modules/arch.fish:abbr syua 'pikaur -Syua'
```

NOTE for bash users: instead of `abbr` use `alias` instead

**Reply by AKorezin** [reply, marked as answer]
Thanks, I've patched it on my side.

**Reply by Draghmar** [reply]
So what there are? When someone comes and tells that suddenly there's an issue with having "useless" app taking tab-completition then maybe the issue is there, don't you think? I was quite surprised when mine stopped working because some weird app took over something that worked for so long. And now the "solution" is to make "hack" changes to each instance where I have pikaur installed... :/ And then we'll to do it for each app you will split? So one change from your side, a lot for each user...

**Reply by AKorezin** [reply]
> And now the "solution" is to make "hack" changes to each instance where I have pikaur installed...

Hi @Draghmar. I can't agree with your position because @actionless is not a supplier, unless you have a contract that says otherwise. You have no right to indignantly demand anything from the author, but I think you can politely ask and suggest something.
If you can afford to sponsor the project, you'll be able to participate in decision making and task prioritization.

**Reply by sehraf** [reply]
> So what there are? When someone comes and tells that suddenly there's an issue with having "useless" app taking tab-completition then maybe the issue is there, don't you think? I was quite surprised when mine stopped working because some weird app took over something that worked for so long. And now the "solution" is to make "hack" changes to each instance where I have pikaur installed... :/ And then we'll to do it for each app you will split? So one change from your side, a lot for each user...

I agree with you, but @AKorezin does have a point.

My solution: https://gist.github.com/sehraf/85ba23b5f6e71b60cb4916ec0bd9b733
drop it in `/usr/share/libalpm/hooks/`

**Reply by Draghmar** [reply]
> > And now the "solution" is to make "hack" changes to each instance where I have pikaur installed...
> 
> Hi @Draghmar. I can't agree with your position because @actionless is not a supplier, unless you have a contract that says otherwise. You have no right to indignantly demand anything from the author, but I think you can politely ask and suggest something. If you can afford to sponsor the project, you'll be able to participate in decision making and task prioritization.

Well...I can. Funny fact - no users for the free app, no potential sponsors and app will die, like many did.

Anyway, my response was directed at the "because I said so" attitude, that way too often happens at apps like this. There are few threads about pikasay change being annoying, with polite explanations but I found response to them way to absolute.

Sigh...Maybe I overreacted a little bit. Sorry @actionless . It's probably because I was caught off-guard with this app, not knowing where it came from at the beginning and then learning that my muscle memory had to change for some reasons.
